### PR TITLE
DataImporterTask should not cast to ICsvDynamicSeparatorSelector but rely on the DI mechanism

### LINKED ISSUE #1213: DataImporterTask should not cast to ICsvDynamicSeparatorSelector but rely on the DI mechanism

### PR DESCRIPTION
**Pull request by abdelr**
Fixes #1213 (DataImporterTask should not cast to ICsvDynamicSeparatorSelector but rely on the DI mechanism)

### CODE REVIEW

**Comment by abdelr** [comment]
Methods are already covered by tests...

File: src/OSPSuite.R/RRegister.cs [hunk 1]
```diff
@@ -44,7 +44,7 @@ public override void RegisterInContainer(IContainer container)
          container.Register<IOSPSuiteExecutionContext, RExecutionContext>(LifeStyle.Singleton);
          container.Register<IOSPSuiteLogger, RLogger, RLogger>(LifeStyle.Singleton);
          container.Register<IEventPublisher, EventPublisher>(LifeStyle.Singleton);
-         container.Register<ICsvSeparatorSelector, CsvSeparatorSelector>(LifeStyle.Singleton);
+         container.Register<ICsvDynamicSeparatorSelector, ICsvSeparatorSelector, CsvSeparatorSelector>(LifeStyle.Singleton);
```
Comment (msevestre) [inline src/OSPSuite.R/RRegister.cs:47]: you need to Exclude from auto registration no?

File: src/OSPSuite.R/Services/DataImporterTask.cs [hunk 2]
```diff
@@ -56,7 +56,7 @@ IContainer container
          //Should actually ask for ICsvDynamicSeparatorSelector as a dependency but I do not currently know
          //how to serve two interfaces with the same singleton. If there is no way to do it with the current
```
Comment (msevestre) [inline src/OSPSuite.R/Services/DataImporterTask.cs:57]: remove comments

**Review by msevestre — CHANGES_REQUESTED** [review]

File: src/OSPSuite.R/RRegister.cs (hunk 1 above)
Comment (abdelr) [reply src/OSPSuite.R/RRegister.cs:48]: It was working fine without Excluding the type. I have just explicitly exclude it but not sure if necessary.

File: src/OSPSuite.R/Services/DataImporterTask.cs (hunk 2 above)
Comment (abdelr) [reply src/OSPSuite.R/Services/DataImporterTask.cs]: Done

File: src/OSPSuite.R/RRegister.cs (hunk 1 above)
Comment (msevestre) [reply src/OSPSuite.R/RRegister.cs:48]: oh it's in R only? I don't think there is automatic registration tehre

Comment (msevestre) [reply src/OSPSuite.R/RRegister.cs:48]: oh yes there is. Then it would have created an issue when getting  ICsvSeparatorSelector (at least in theory)

**Pull request merged**